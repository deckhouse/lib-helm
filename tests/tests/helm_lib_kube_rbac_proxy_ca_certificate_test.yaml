suite: helm_lib_kube_rbac_proxy_ca_certificate definition
templates:
  - helm_lib_kube_rbac_proxy_ca_certificate.yaml
tests:
  - it: renders configmap 'kube-rbac-proxy-ca.crt' in passed namespace

    set:
      global:
        internal:
          modules:
            kubeRBACProxyCA:
              cert: cert

    asserts:
      - containsDocument:
          kind: ConfigMap
          apiVersion: v1
          name: kube-rbac-proxy-ca.crt
          namespace: d8-ns

  - it: contains certificate in 'ca.crt' data key

    set:
      global:
        internal:
          modules:
            kubeRBACProxyCA:
              cert: |
                -----BEGIN CERTIFICATE-----
                Cert
                -----END CERTIFICATE-----

    asserts:
      - equal:
          path: 'data["ca.crt"]'
          value: |
            -----BEGIN CERTIFICATE-----
            Cert
            -----END CERTIFICATE-----

  # this assertion is example for match all document
  - it: does not change document between test runs

    set:
      global:
        internal:
          modules:
            kubeRBACProxyCA:
              cert: cert

    asserts:
      - matchSnapshot: {}