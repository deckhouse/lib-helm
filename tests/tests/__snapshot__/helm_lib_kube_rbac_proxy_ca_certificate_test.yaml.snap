does not change document between test runs:
  1: |
    apiVersion: v1
    data:
      ca.crt: |2
        cert
    kind: ConfigMap
    metadata:
      annotations:
        kubernetes.io/description: |
          Contains a CA bundle that can be used to verify the kube-rbac-proxy clients.
      labels:
        heritage: deckhouse
        module: test-module
      name: kube-rbac-proxy-ca.crt
      namespace: d8-ns
